phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java

```diff
@@ -454,8 +455,10 @@ protected static String checkClusterInitialized(ReadOnlyProps serverProps) throw
     protected static String setUpTestCluster(@Nonnull Configuration conf, ReadOnlyProps overrideProps) throws Exception {
         boolean isDistributedCluster = isDistributedClusterModeEnabled(conf);
         if (!isDistributedCluster) {
+            TEARDOWN_THRESHOLD.set(30);
             return initMiniCluster(conf, overrideProps);
         } else {
+            TEARDOWN_THRESHOLD.set(1);
```

Comment:
Do we need to clear up after each test ?
A Real cluster should have enough resources to hold some leftover tables.

Reply:
I'm also not conviced that needs to be an AtomicInteger.

Reply:
> Do we need to clear up after each test ? A Real cluster should have enough resources to hold some leftover tables.

The cleanup logic is more simpler. Else we will need to track the tables created per jvm and then clean them up based on revised thresholds.

Reply:
Why would we need to track the tables by JVM ?
These test don't run in parallel, do they ?